integration/users/update_user_test.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package users
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+
+	"github.com/FerretDB/FerretDB/integration"
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
+)
+
+func TestUpdateUser(t *testing.T) {
+	t.Parallel()
+
+	ctx, collection := setup.Setup(t)
+	db := collection.Database()
+
+	testCases := map[string]struct { //nolint:vet // for readability
+		createPayload bson.D
+		updatePayload bson.D
+
+		expected   bson.D
+		err        *mongo.CommandError
+		altMessage string
+
+		skipForMongoDB string
+	}{
+		"MissingFields": {
+			createPayload: bson.D{
+				{"createUser", "missing_fields"},
+				{"roles", bson.A{}},
+				{"pwd", "pass123654"},
+			},
+			updatePayload: bson.D{
+				{"updateUser", "missing_fields"},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "Must specify at least one field to update in updateUser",
+			},
+		},
+		"UserNotFound": {
```

Comment:
Do we need a separate test where username is set as an empty string when update is called? Or is this one enough for this case too?

Reply:
Good idea!